subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/DarwinFileWatcherRegistryFactory.java

```diff
@@ -51,7 +53,7 @@ protected FileWatcherUpdater createFileWatcherUpdater(
         FileWatcherProbeRegistry probeRegistry,
         WatchableHierarchies watchableHierarchies
     ) {
-        return new HierarchicalFileWatcherUpdater(watcher, DarwinFileWatcherRegistryFactory::validateLocationToWatch, probeRegistry, watchableHierarchies, root -> root);
+        return new HierarchicalFileWatcherUpdater(watcher, DarwinFileWatcherRegistryFactory::validateLocationToWatch, probeRegistry, watchableHierarchies, (SnapshotHierarchy root) -> Collections.emptyList());
```

Comment:
Why do we need to specify the type here now?

Reply:
We don't. Removed.